### ISSUE TITLE
Add support to retrieve an object based on the resource id

### ISSUE THREAD

**Issue by LaurentLesle** [Azure/terraform-provider-azapi#134]
This request is to align the data.azapi_resource on a similar behaviour we already have with the az cli.

The use case is a public ssh key is in a remote subscription and I want to reference to create a linux virtual machine

```
# Get an sshkey from the resource id
az sshkey show --ids "/subscriptions/xxxxxxx/resourceGroups/yyyy/providers/Microsoft.Compute/sshPublicKeys/ssh-key"
```

Proposed implementation
```
provider "azapi" {
}

variable "ssh_public_key_id" {
  default = "/subscriptions/xxxxxxx/resourceGroups/yyyy/providers/Microsoft.Compute/sshPublicKeys/ssh-key"
}

data "azapi_resource" "ssh_public_key" {
  id    = var.ssh_public_key_id
  type  = "Microsoft.Compute/sshPublicKeys@2021-11-01"

  response_export_values = ["publicKey"]
}

output "ssh_public_key" {
   value = jsondecode(data.azapi_resource.ssh_public_key.output).publicKey
}
```

With this proposition, the name and parent_id are optional when id is set.

**Comment by ms-henglu** [comment]
Hi @LaurentLesle ,

Thank you for opening this issue. I'll add this feature in next release.

**Comment by ms-henglu** [comment]
Hi @LaurentLesle,

This feature has been released in v0.4.0, would you please give it a try? Thanks!

**Comment by daveneeley** [comment]
`resource_id` won't validate for a resource like [Web Apps - List Host Keys Slot][1], which does not have a number of paths divisible by 2 (there are 13). Since name cannot be provided when resource_id is, there is no way to add this additional information.

[1]: https://docs.microsoft.com/en-us/rest/api/appservice/web-apps/list-host-keys-slot
